src/System.Runtime/tests/System/Text/ASCIIUtilityTests.netcoreapp.cs

```diff
@@ -0,0 +1,460 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Buffers;
+using System.Numerics;
+using System.Reflection;
+using System.Runtime.InteropServices;
+using System.Security.Cryptography;
+using Xunit;
+
+namespace System.Text.Tests
+{
+    // Since many of the methods we'll be testing are internal, we'll need to invoke
+    // them via reflection.
+    public static unsafe partial class AsciiUtilityTests
+    {
+        private const int SizeOfVector128 = 128 / 8;
+
+        // The delegate definitions and members below provide us access to CoreLib's internals.
```

Comment:
We also have a project at https://github.com/dotnet/corefx/blob/master/src/Common/tests/Common.Tests.csproj that's used to test various common helpers.  You could put these tests there instead of using reflection.

Reply:
Interesting. Do you know how that test project works when faced with shared source files that contain `#if BIT64` and similar?

Reply:
It probably doesn't. :)

Reply:
The curse of `nuint` strikes again. 😈